### ISSUE TITLE
Add telemetry events

### ISSUE THREAD

**Issue by danschultzer** [danschultzer/pow#492]
This was already discussed way back in https://github.com/danschultzer/pow/issues/122, but now with the [Phoenix Live Dashboard](https://github.com/phoenixframework/phoenix_live_dashboard) announcement I think it's time to add in the telemetry events so it can easily be hooked up.

OWASP [recommends logging the following](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Session_Management_Cheat_Sheet.md#logging-sessions-life-cycle-monitoring-creation-usage-and-destruction-of-session-ids):

- Session ID creation
- Session ID renewal
- Session ID destruction
- Login and logout operations 
- Privilige changes
- Timeout expiration
- Invalid session activities
- Critical business operations

I think the following could be nice metrics:

- Number of current sessions
- Number of users signed in
- Number of failed sign in attempts
- Average number of failed attempts per user
- Average number of sessions per user
- Average authentication duration time

Please share if you got any suggestions for what to track 😄

**Comment by steffenix** [comment]
I would only add  `total numbers of users`.

**Comment by cdvv7788** [comment]
Registrations would be useful for this too? Or this is meant to track only sessions?
For the extensions, things like confirmed accounts could be useful too.

**Comment by danschultzer** [comment]
> Registrations would be useful for this too? Or this is meant to track only sessions?

Yeah, since registrations is just collecting what exists in the DB at any certain point.

For total number of signed in users, this is great: https://ghost.jmnorlund.net/instrumenting-an-elixir-phoenix-pow-application/ (by @jmn)
